### PR TITLE
Re-enable some analyzers

### PR DESCRIPTION
**Pull request by JeremyKuhne**
There were a few marked as TODO. This makes them back into warnings with the exception of IDE0059 which is "Unnecessary assignment of a value". That is now a suggestion. Some of the fixes for IDE0059 are in this commit, others will follow as there are a lot of failures on that one.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12529)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
src\System.Windows.Forms.Design\tests\UnitTests\GlobalSuppressions.cs(6,83): error SA1404: (NETCORE_ENGINEERING_TELEMETRY=Build) Code analysis suppression should have justification

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs [hunk 1]
```diff
@@ -136,27 +136,51 @@ public override DesignerActionItemCollection GetSortedActionItems()
         DesignerActionItemCollection items = [];
         if (!IsReadOnly)
         {
-            items.Add(new DesignerActionMethodItem(this, "InvokeEmbedVerb", SR.ToolStripDesignerEmbedVerb, "", SR.ToolStripDesignerEmbedVerbDesc, true));
+            items.Add(new DesignerActionMethodItem(
+                this,
+                nameof(InvokeEmbedVerb),
+                SR.ToolStripDesignerEmbedVerb,
+                "",
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs:143]:
```suggestion
                string.Empty,
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs [hunk 2]
```diff
@@ -136,27 +136,51 @@ public override DesignerActionItemCollection GetSortedActionItems()
...
         if (CanAddItems)
         {
             if (_toolStrip is not StatusStrip)
             {
-                items.Add(new DesignerActionMethodItem(this, "InvokeInsertStandardItemsVerb", SR.ToolStripDesignerStandardItemsVerb, "", SR.ToolStripDesignerStandardItemsVerbDesc, true));
+                items.Add(new DesignerActionMethodItem(
+                    this,
+                    nameof(InvokeInsertStandardItemsVerb),
+                    SR.ToolStripDesignerStandardItemsVerb,
+                    "",
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs:156]:
```suggestion
                    string.Empty,
```

**Review by Tanya-Solyanik — APPROVED** [review]
added minor comments

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/ToolStripActionList.cs:156]: Before .NET Framework 2.0 this made a difference for performance. Now they are equivalent, and up to the dev to decide which is more appropriate for their case.

**Pull request merged**